### PR TITLE
Feature/dp 438 international identifier register organisation

### PR DESCRIPTION
**Pull request by shilpigoel-goaco**

### CODE REVIEW

File: Frontend/CO.CDP.OrganisationApp/Pages/Registration/OrganisationIdentification.cshtml [hunk 1]
```diff
@@ -27,142 +27,147 @@
...
+                        <h1 class="govuk-fieldset__heading">
+                            @ViewData["Title"]
+                        </h1>
+                    </legend>
+                    <div id="organisationidentification-hint" class="govuk-hint">
+                        @StaticTextResource.Organisation_OrganisationIdentification_Hint
+                    </div>
+                    <p>
+                        <a class="govuk-link govuk-!-font-size-19" href="/registration/organisation-identification-country">
+                            Enter a non-UK registry and number
```
Comment (andymantell) [inline Frontend/CO.CDP.OrganisationApp/Pages/Registration/OrganisationIdentification.cshtml:50]: This string should be localized

File: Frontend/CO.CDP.OrganisationApp/Pages/Registration/OrganisationInternationalIdentification.cshtml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            <div class="govuk-form-group">
+                <fieldset class="govuk-fieldset">
+                    <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
+                        <h1 class="govuk-fieldset__heading">
+                            @ViewData["Title"]
+                        </h1>
+                    </legend>
+
+                    <div id="organisationidentification-hint" class="govuk-hint">
+                        Select one then enter the registration number
```
Comment (andymantell) [inline Frontend/CO.CDP.OrganisationApp/Pages/Registration/OrganisationInternationalIdentification.cshtml:38]: This string should be localized

File: Frontend/CO.CDP.OrganisationApp/Pages/Registration/OrganisationInternationalIdentification.cshtml [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                                </div>
+
+                                <div class="govuk-radios__conditional @(Model.OrganisationScheme == choice.Scheme ? "" : "govuk-radios__conditional--hidden")" id="conditional-@id">
+                                    <div class="govuk-form-group @(hasError ? "govuk-form-group--error" : "")">
+                                        @Html.LabelFor(m => m.RegistrationNumbers, htmlAttributes: new { @class = "govuk-label" })
+                                        @if (hasError)
+                                        {
+                                            <span id="RegistrationNumber-error" class="govuk-error-message">
+                                                <span class="govuk-visually-hidden">@StaticTextResource.Global_Error:</span>
+                                                @ModelState[nameof(Model.RegistrationNumbers)]!.Errors[0].ErrorMessage
```
Comment (andymantell) [inline Frontend/CO.CDP.OrganisationApp/Pages/Registration/OrganisationInternationalIdentification.cshtml:68]:
```suggestion
                                                @Html.ValidationMessageFor(m => m.RegistrationNumbers)
```

File: Frontend/CO.CDP.OrganisationApp/Pages/Registration/OrganisationInternationalIdentification.cshtml [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                        </div>
+                        <div class="govuk-radios__conditional @(Model.OrganisationScheme == $"{Model.Country}-Other" ? "" : "govuk-radios__conditional--hidden")" id="conditional-other">
+                            <div class="govuk-form-group @(ModelState[nameof(Model.RegistrationNumbers)]?.Errors?.Any() ?? false ? "govuk-form-group--error" : "")">
+                                @Html.LabelFor(m => m.RegistrationNumbers, htmlAttributes: new { @class = "govuk-label" })
+
+                                @if (ModelState[nameof(Model.RegistrationNumbers)]?.Errors?.Any() ?? false)
+                                {
+                                    <span id="RegistrationNumber-error" class="govuk-error-message">
+                                        <span class="govuk-visually-hidden">@StaticTextResource.Global_Error:</span>
+                                        @ModelState[nameof(Model.RegistrationNumbers)]!.Errors[0].ErrorMessage
```
Comment (andymantell) [inline Frontend/CO.CDP.OrganisationApp/Pages/Registration/OrganisationInternationalIdentification.cshtml:106]:
```suggestion
                                        @Html.ValidationMessageFor(m => m.RegistrationNumbers)
```

File: Frontend/CO.CDP.OrganisationApp/Pages/Registration/OrganisationInternationalIdentificationCountry.cshtml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                    <div class="govuk-form-group @(countryHasError ? "govuk-form-group--error" : "")">
+                        @if (countryHasError)
+                        {
+                            <p id="country-error" class="govuk-error-message">
+                                <span class="govuk-visually-hidden">Error:</span>
+                                @ModelState[nameof(Model.Country)]!.Errors[0].ErrorMessage
+                            </p>
+                        }
+                        <select asp-for="Country" class="govuk-select @(countryHasError ? "govuk-select--error" : "")" aria-describedby="country-error">
+                            <option value="">Please select</option>
```
Comment (andymantell) [inline Frontend/CO.CDP.OrganisationApp/Pages/Registration/OrganisationInternationalIdentificationCountry.cshtml:44]:
```suggestion
                            <option value="">@StaticTextResource.Global_PleaseSelect</option>
```

File: Libraries/CO.CDP.Mvc.Validation/RequiredIfAttribute.cs [hunk 6]
```diff
@@ -23,6 +23,43 @@ protected override ValidationResult IsValid(object? value, ValidationContext val
...
+                if (string.IsNullOrEmpty(registrationNumberValue))
+                {                    
+                    return new ValidationResult(errorMessage ?? $"{validationContext.DisplayName} is required");
+                }
+            }
+        }
+        else if (!string.IsNullOrEmpty(organisationSchemeValue) && value == null)
+        {
+            return new ValidationResult(errorMessage ?? $"{validationContext.DisplayName} is required");
+        }
```
Comment (andymantell) [inline Libraries/CO.CDP.Mvc.Validation/RequiredIfAttribute.cs:57]: This is some very specific code relating to registration numbers added to a very generic `RequiredIf` validator which is used in many places - I don't think this code should live here. Can you either create a validation attribute specific to this use case, or handle the validation in the controller directly?

File: Services/CO.CDP.Localization/StaticTextResource.resx [hunk 7]
```diff
@@ -285,9 +285,6 @@
   <data name="Organisation_MyOrganisations" xml:space="preserve">
     <value>My organisations</value>
   </data>
-  <data name="Organisation_OrganisationName" xml:space="preserve">
```
Comment (andymantell) [inline Services/CO.CDP.Localization/StaticTextResource.resx:288]: Why have all these values been removed?

File: Services/CO.CDP.Localization/StaticTextResource.resx [hunk 8]
```diff
@@ -2038,4 +2011,28 @@
   <data name="Exclusions_AddExclusionDeclaration_DiscretionaryExclusionsLinkText" xml:space="preserve">
     <value>These exclusions are defined in &lt;a href="{0}" target="_blank"&gt;schedule 7 of the Procurement Act 2023 (opens in new tab)&lt;/a&gt;.</value>
   </data>
+  <data name="OrganisationRegistration_InternationalCountry_Heading" xml:space="preserve">
+    <value>Select which country your organisation is registered</value>
+  </data>
+  <data name="Organisation_International_Country_Heading" xml:space="preserve">
+    <value>Select which country your organisation is registered</value>
+  </data>
```
Comment (andymantell) [inline Services/CO.CDP.Localization/StaticTextResource.resx:2019]: Should this say "Select which country your organisation is registered _in_" ?

**Review by andymantell — CHANGES_REQUESTED** [review]

File: Services/CO.CDP.Localization/StaticTextResource.resx (hunk 7 above)
Comment (shilpigoel-goaco) [reply Services/CO.CDP.Localization/StaticTextResource.resx:288]: It is not removed, I just checked and it is there.

File: Services/CO.CDP.Localization/StaticTextResource.resx (hunk 8 above)
Comment (andymantell) [inline Services/CO.CDP.Localization/StaticTextResource.resx:2019]: These look like duplicates?

Comment (andymantell) [reply Services/CO.CDP.Localization/StaticTextResource.resx:2019]: Also, the two resources seem to be duplicated?

File: Services/CO.CDP.Localization/StaticTextResource.resx (hunk 7 above)
Comment (andymantell) [reply Services/CO.CDP.Localization/StaticTextResource.resx:288]: Ah! They were duplicated on main and you've removed the duplicates. Cool :-)

**Review by andymantell — CHANGES_REQUESTED** [review]

**Review by andymantell — APPROVED** [review]

File: Services/CO.CDP.Localization/StaticTextResource.resx [hunk 9]
```diff
@@ -2038,4 +2011,34 @@
   <data name="Exclusions_AddExclusionDeclaration_DiscretionaryExclusionsLinkText" xml:space="preserve">
     <value>These exclusions are defined in &lt;a href="{0}" target="_blank"&gt;schedule 7 of the Procurement Act 2023 (opens in new tab)&lt;/a&gt;.</value>
   </data>
+  <data name="OrganisationRegistration_InternationalCountry_Heading" xml:space="preserve">
+    <value>Select which country your organisation is registered</value>
+  </data>
+  <data name="OrganisationRegistration_InternationalIdentifier_Country_Heading" xml:space="preserve">
+    <value>Select which country your organisation is registered</value>
+  </data>
```
Comment (andymantell) [inline Services/CO.CDP.Localization/StaticTextResource.resx:2019]: These two still seem to be duplicated. And still don't make gramatical sense?

`Select which country your organisation is registered` should be `Select which country your organisation is registered in`

**Review by andymantell — APPROVED** [review]

**Comment by shilpigoel-goaco** [comment]
work merged within DP-831 ticket PR.

**Pull request closed without merging**